microsoft/testsuites/network/netinterface.py

```diff
@@ -0,0 +1,119 @@
+from lisa.util import constants
+
+
+@TestSuiteMetadata(
+    area="network",
+    category="functional",
+    description="""
+    This test suite validates basic functionalities of Network Interfaces.
+    """,
+)
+class NetInterface(TestSuite):
+    RELOAD_NETVSC_TEST_COUNT = 10
+    NET_IFUP_IFDOWN_TEST_COUNT = 4
+
+    @TestCaseMetadata(
+        description="""
+            This test case verifies if synthetic network interface can be
+            brought up and brought down gracefully.
+
+            Steps:
+            1. Validate netvsc isn't built-in already. If it is then skip the test.
+            2. Unload and load netvsc module multiple times in a loop.
+            3. Change nic state to up and down multiple times.
+                Each time after "up" state, verify ip address is assigned to nic
+                and internet is accessible via nic.
+
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            min_nic_count=1,
+            network_interface=schema.NetworkInterfaceOptionSettings(
+                data_path=schema.NetworkDataPath.Synthetic,
+            ),
+        ),
+    )
+    def validate_network_interface_ifup_ifdown(self, node: Node, log: Logger) -> None:
+        self._validate_netvsc_built_in(node)
+        network_interface_feature = node.features[NetworkInterface]
+
+        # Test loading and unloading netvsc driver
+        test_count = 0
+        while test_count < self.RELOAD_NETVSC_TEST_COUNT:
+            test_count += 1
+            # Unload hv_netvsc
+            network_interface_feature.reload_netvsc()
+
+        # Clean all dhclient processes, get IP & try ping
+        node_nic_info = Nics(node)
+        node_nic_info.initialize()
+        default_nic = node_nic_info.default_nic
+        default_route = node_nic_info.default_nic_route
+
+        node_nic_info.change_nic_state(default_nic, "up")
+        if not node_nic_info.nics[default_nic].ip_addr:
+            node.execute("kill $(pidof dhclient)", shell=True, sudo=True)
+            dhclient = node.tools[Dhclient]
+            dhclient.release()
+            dhclient.renew()
+            sleep(15)
+
+        self._ping_check
+
+        test_count = 0
+        while test_count < self.NET_IFUP_IFDOWN_TEST_COUNT:
+            self.__log.info(f"NET IFUP IFDOWN Test Iteration : {test_count}")
+            node_nic_info.change_nic_state(default_nic, "down")
+            node_nic_info.change_nic_state(default_nic, "up")
+            if not node_nic_info.default_nic:
+                # Add default route if missing after running ip link down/up
+                node.execute(f"ip route add {default_route}", shell=True, sudo=True)
+
+            if not node_nic_info.nics[default_nic].ip_addr:
+                node.execute("kill $(pidof dhclient)", shell=True, sudo=True)
+                dhclient = node.tools[Dhclient]
+                dhclient.release()
+                dhclient.renew()
+                sleep(15)
```

Comment:
Don't put hardcode sleep, use a condition, and check it with interval like 0.5 or 1 second.

Reply:
ok will change here. Just note to add, we are already using hard-coded sleeps in dpdksuite and kdumpcrash tests.

Reply:
It's ok to have longer timeout even one minute, but we need to check it every 1 or 2 second. There may be a way to check it by dhclient.